packages/wonder-blocks-dropdown/components/dropdown-core.js

```diff
@@ -93,30 +144,78 @@ export default class DropdownCore extends React.Component<DropdownCoreProps> {
         }
     };
 
+    getTranslatedPosition() {
+        if (this.props.alignment === "right" ) {
+            return {
+                transform: `translate3d(${-(this.menuWidth || 0)}px, 0, 0)`,
+            }
+        }
+    }
+
+    renderMenu(outOfBoundaries: ?boolean) {
+        const {items, light, style} = this.props;
+
+        const translated = this.getTranslatedPosition();
+        return <View
+            onMouseUp={(event) => {
+                // Stop propagation to prevent the mouseup listener
+                // on the document from closing the menu.
+                event.nativeEvent.stopImmediatePropagation();
+            }}
+            ref={node => this.menuElement = ((ReactDOM.findDOMNode(node): any): Element)}
+            style={[styles.dropdown, light && styles.light, translated, outOfBoundaries && styles.hidden, style]}
+        >
+            {items}
+        </View>;
+    }
+
+    renderDropdown() {
+        const {alignment, openerElement} = this.props;
+        // If we are in a modal, we find where we should be portalling the menu
+        // by using the helper function from the modal package on the opener
+        // element.
+        // If we are not in a modal, we use body as the location to portal to.
+        const modalHost = maybeGetPortalMountedModalHostElement(openerElement) || document.querySelector("body");
+
+        if (modalHost) {
+            return ReactDOM.createPortal(
+                <Popper
+                    referenceElement={this.props.openerElement}
+                    // TODO(sophie): allow dropdown to be a dropup (rise-up?) --
+                    // to appear above the dropdown opener
+                    placement={alignment === "left" ? "bottom-start" : "bottom-end"}
```

Comment:
My impression is that `bottom-end` is supposed to right-align the elements, but instead the menu starts where the opener ends. So I move it back left by the width of the menu by adding another translate via `getTranslatedPosition`.

Reply:
@somewhatabstract did you encounter this issue with the tooltips? It seems like an odd case!

Reply:
I think there's a misinterpretation of the values happening here. That certainly shouldn't be necessary. Instead, there's a `flip` modifier that auto detects the position based off the bounds element. There should be no need to do all this extra work.

Reply:
I figured it out -- I was positioning a child element with `absolute` when I didn't need to. Thanks for helping me find my error!